klayout/drc/rule_decks/mim_b.drc

```diff
@@ -0,0 +1,149 @@
+  mimtm4_l1.forget
+  mimtm4_l2.forget
+  mimtm4_l.forget
+
+  # Rule MIMTM.5: Minimum spacing between top plate and the Vian-1 connecting to the bottom plate is 0.4µm.
+  logger.info('Executing rule MIMTM.5')
+  mimtm5_l1 = fusetop.separation(top_via.interacting(topmin1_metal), 0.4.um, euclidian)
+  mimtm5_l1.output('MIMTM.5',
+                   'MIMTM.5 : Minimum spacing between top plate and the Vian-1 connecting to the bottom plate: 0.4µm')
+  mimtm5_l1.forget
+
+  # Rule MIMTM.6: Minimum spacing between unrelated top plates is 0.6µm.
+  logger.info('Executing rule MIMTM.6')
+  mimtm6_l1 = fusetop.space(0.6.um, euclidian)
+  mimtm6_l1.output('MIMTM.6', 'MIMTM.6 : Minimum spacing between unrelated top plates: 0.6µm')
+  mimtm6_l1.forget
+
+  # Rule MIMTM.7: Min FuseTop enclosure by CAP_MK is 0um.
+  logger.info('Executing rule MIMTM.7')
+  mimtm7_l1 = fusetop.not(cap_mk)
+  mimtm7_l1.output('MIMTM.7', 'MIMTM.7 : Min FuseTop enclosure by CAP_MK: 0um')
+  mimtm7_l1.forget
+
+  # Rule MIMTM.8a: Minimum MIM cap area (defined by FuseTop area) is 25µm².
+  logger.info('Executing rule MIMTM.8a')
+  mimtm8a_l1 = fusetop.with_area(nil, 25.um)
+  mimtm8a_l1.output('MIMTM.8a', 'MIMTM.8a : Minimum MIM cap area (defined by FuseTop area): 25µm²')
+  mimtm8a_l1.forget
+
+  # Rule MIMTM.8b: Maximum single MIM Cap area (Use multiple MIM caps in
+  ## parallel connection if bigger capacitors are required) (um2) is 10000µm.
+  logger.info('Executing rule MIMTM.8b')
+  mimtm8b_l1 = fusetop.with_area(10_000.um, nil).not_in(fusetop.with_area(10_000.um))
+  mimtm8b_l1.output('MIMTM.8b', "MIMTM.8b : Maximum single MIM Cap area (Use multiple MIM caps in
+                     parallel connection if bigger capacitors are required) (um2): 10000µm")
+  mimtm8b_l1.forget
+
+  # Rule MIMTM.9: Min. Via (Vian-1) spacing for sea of Via on MIM top plate is 0.5µm.
+  logger.info('Executing rule MIMTM.9')
+  mimtm9_l1 = top_via.inside(fusetop).space(0.5.um, euclidian)
+  mimtm9_l1.output('MIMTM.9', 'MIMTM.9 : Min. Via (Vian-1) spacing for sea of Via on MIM top plate: 0.5µm')
+  mimtm9_l1.forget
+
+  # Rule MIMTM.10: (a) There cannot be any Vian-2 touching MIM bottom plate Metaln-1.
+  ## (b) MIM bottom plate Metaln-1 can only be connected through the higher Via (Vian-1).
+  logger.info('Executing rule MIMTM.10')
+  mimtm10_l1 = topmin1_via.interacting(topmin1_metal_fusetop)
+  mimtm10_l1.output('MIMTM.10', "MIMTM.10 : (a) There cannot be any Vian-2 touching MIM bottom plate Metaln-1.
+                    (b) MIM bottom plate Metaln-1 can only be connected through the higher Via (Vian-1).")
+  mimtm10_l1.forget
+
+  # Rule MIMTM.11: Bottom plate of multiple MIM caps can be shared (for common nodes)
+  ## as long as total MIM area with that single common plate does not exceed MIMTM.8b rule.
+  logger.info('Executing rule MIMTM.11')
+  mimtm11_large_topmin1_metal = topmin1_metal_fusetop.with_area(10_000, nil)
```

Comment:
shouldn't we reuse the thing computed in `MIMTM.8b` rather than recalcultate them here?

Reply:
@proppy `MIMTM.8b` is to check MIM top plate which is defined by `fusetop` layer. For `MIMTM.11`, it's an another check that intended for MIM bottom plate. So we couldn't use same layer in both rules.

Reply:
ah I see, sorry I missed this, do you think it would be useful to have a different naming conversion between variable that are local to a rule and other that are shared between multiple rules?

Reply:
@proppy I'm not sure I get the request that you are asking here. MTM.8b and MTMTM.11 are different rules, yes they share the rule wording and value. But they are implemented on 2 different layers. I'm not sure what you are looking for here?

Reply:
sorry for the lack of clarity in my comment, I meant that layers like  `topmin1_metal_fusetop` or `mimtm_virtual` are used by many rules whereas other layer like `mimtm11_large_topmin1_metal` are specific to a given rule.

I wonder if it would make sense to adopt a naming convention to differentiate between the two (currently we only prefix rule specific layer, but would it make sense to upcase or prefix global layer similarly?); or even better implement each rule in a separate function w/ a block so that it can be explicit the scoping (local layer would be local and forgot automatically, and global variable would need to be passed as explicit parameters)

Reply:
filed https://github.com/google/globalfoundries-pdk-libs-gf180mcu_fd_pv/issues/77 to discuss this further.